content/en/docs/tasks/network/extend-service-ip-ranges.md

```diff
@@ -196,3 +200,90 @@ kubectl get servicecidr newcidr1
 ```
 Error from server (NotFound): servicecidrs.networking.k8s.io "newcidr1" not found
 ```
+
+## Kubernetes Service CIDR Policies
+
+Cluster administrators can implement policies to control the creation and
+modification of ServiceCIDR resources within the cluster. This allows for
+centralized management of the IP address ranges used for Services and helps
+prevent unintended or conflicting configurations. Kubernetes provides mechanisms
+like Validating Admission Policies to enforce these rules.
+
+### Preventing Unauthorized ServiceCIDR Creation/Update using Validating Admission Policy
+
+There can be situations that the cluster administrators want to restrict the
+ranges that can be allowed or to completely deny any changes to the cluster
+Service IP ranges.
+
+{{< note >}}
+The default "kubernetes" ServiceCIDR is created by the kube-apiserver
+to provide consistency in the cluster and is required for the cluster to work,
+so it always must be allowed.
+{{</ note >}}
+
+#### Restrict Service CIDR ranges to some specific ranges
+
+The following example demonstrates how to use a `ValidatingAdmissionPolicy` and
+its binding to restrict the Service CIDR ranges that can be created or updated,
+excluding the default "kubernetes" ServiceCIDR:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "servicecidrs.default"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["networking.k8s.io"]
+      apiVersions: ["v1","v1beta1"]
+      operations:  ["CREATE", "UPDATE"]
```

Comment:
does UPDATE actually matter, given that `spec.cidrs` is immutable?

Also, is the `v1beta1` necessary? If so, do we need `v1alpha1` too?

Reply:
I made this also for our internal use cases, we have users in prod with v1beta, I'm not aware of anyone using it in alpha 😄

Reply:
The important question to me is not whether anyone is _currently_ using those api versions, it's whether people in the future can use those api versions to bypass VAPs. Are _all_ clusters required to say `apiVersions: ["v1", "v1beta1"]` (or `apiVersions: ["v1", "v1beta1", "v1alpha1"]`) to prevent people from bypassing the VAP by creating a ServiceCIDR with an older apiVersion? (If yes, this seems like a terrible flaw in VAP. If no, then you shouldn't need to specify `"v1beta1"` here, unless you're doing it for people running versions of k8s that don't have the `"v1"` API yet, but in that case, would the VAP even be accepted?)

Reply:
> it's whether people in the future can use those api versions to bypass VAPs

oh, understand, sorry,  alpha is not served so it can not be used, the apiserver does not understand it.

> Are all clusters required to say

I think this is the same as webhooks, and you can say any apiversions with the asterisk.
Other thing is also the RBAC permissions you give to people, in addition to the policies that you apply, and also betas are removed after X releases (6??) I can't remember now.

AFAIK the VAP does not validate against the existing types enabled, it just install rules that match against the request, @jpbetz is the expert

Reply:
UPDATE can change finalizers and annotations and labels, so I think is relevant, since ServiceCIDRs are protected by finalizers

Reply:
the policies will be tested here https://github.com/kubernetes/kubernetes/pull/131318/commits/c666380c04991f98fc9dbd8e1073b6bea8193bc8